test/algorithms/length/length_geo.cpp

```diff
@@ -0,0 +1,144 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+// Unit Test
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fisikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <algorithms/test_length.hpp>
+
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/adapted/std_pair_as_segment.hpp>
+
+#include <test_geometries/all_custom_linestring.hpp>
+#include <test_geometries/wrapped_boost_array.hpp>
+
+template <typename P>
+struct geo_strategies{
+    // Set radius type, but for integer coordinates we want to have floating point radius type
+    typedef typename bg::promote_floating_point
+    <
+    typename bg::coordinate_type<P>::type
+    >::type rtype;
+
+    typedef bg::srs::spheroid<rtype> stype;
+
+    typedef bg::strategy::distance::andoyer<stype> andoyer_type;
+
+    BOOST_CONCEPT_ASSERT
+        (
+            (bg::concept::PointDistanceStrategy<andoyer_type, P, P>)
+        );
+
+    typedef bg::strategy::distance::thomas<stype> thomas_type;
+
+    BOOST_CONCEPT_ASSERT
+        (
+            (bg::concept::PointDistanceStrategy<thomas_type, P, P>)
+        );
+
+    typedef bg::strategy::distance::vincenty<stype> vincenty_type;
+
+    BOOST_CONCEPT_ASSERT
+        (
+            (bg::concept::PointDistanceStrategy<vincenty_type, P, P>)
+        );
+
+    //typedef typename bg::strategy::distance::services::default_strategy
+    //<
+    //    bg::point_tag, bg::point_tag, P
+    //>::type default_strategy_type;
+
+};
+
+std::vector<std::string > Ls_data_gen()
```

Comment:
I do not really see the point of this method.
You can simply define the array and then use Boost.Range functionality to get what you want...